File: frontend/src/utils/phone.js
/** Pakistani mobile: 11 digits, starts with 03 (e.g. 0300 1234567). */

export function phoneDigits(value) {
  return String(value || '').replace(/\D/g, '');
}

export function normalizePakPhone(value) {
  let digits = phoneDigits(value);
  if (digits.startsWith('0092')) {
    digits = `0${digits.slice(4)}`;
  } else if (digits.startsWith('92') && digits.length >= 12) {
    digits = `0${digits.slice(2)}`;
  }
  return digits.slice(0, 11);
}

/** Display format: 03XX XXXXXXX */
export function formatPakPhone(value) {
  const digits = normalizePakPhone(value);
  if (digits.length <= 4) return digits;
  return `${digits.slice(0, 4)} ${digits.slice(4)}`;
}

export function isValidPakPhone(value) {
  return /^03\d{9}$/.test(normalizePakPhone(value));
}

export function validatePakPhone(value, { required = true } = {}) {
  const digits = normalizePakPhone(value);
  if (!digits) {
    return required ? 'Phone number is required.' : null;
  }
  if (digits.length < 11) {
    return 'Enter 11 digits (e.g. 0300 1234567).';
  }
  if (!/^03\d{9}$/.test(digits)) {
    return 'Enter a valid mobile number starting with 03.';
  }
  return null;
}

export const PAK_PHONE_PLACEHOLDER = '0300 1234567';
export const PAK_PHONE_INPUT_MAX_LENGTH = 12;

File: backend/customers/phone.py
import re

PK_MOBILE_RE = re.compile(r'^03\d{9}$')


def normalize_pk_phone(phone):
    digits = re.sub(r'\D', '', str(phone or ''))
    if digits.startswith('0092'):
        digits = f'0{digits[4:]}'
    elif digits.startswith('92') and len(digits) >= 12:
        digits = f'0{digits[2:]}'
    return digits[:11]


def format_pk_phone(phone):
    digits = normalize_pk_phone(phone)
    if len(digits) <= 4:
        return digits
    return f'{digits[:4]} {digits[4:]}'


def validate_pk_phone(phone, *, required=True):
    digits = normalize_pk_phone(phone)
    if not digits:
        return 'Phone number is required.' if required else None
    if len(digits) != 11:
        return 'Enter 11 digits (e.g. 0300 1234567).'
    if not PK_MOBILE_RE.match(digits):
        return 'Enter a valid mobile number starting with 03.'
    return None
